spec/hyacinth/digital_object/type_def/dynamic_field_data_spec.rb

```diff
@@ -0,0 +1,158 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe Hyacinth::DigitalObject::TypeDef::DynamicFieldData do
+  let(:type_def) { described_class.new(:descriptive_metadata) }
+
+  let(:field_definitions) do
+    {
+      dynamic_field_categories: [
+        {
+          display_label: 'Descriptive Metadata',
+          dynamic_field_groups: [
+            {
+              string_key: 'name',
+              display_label: 'Name',
+              dynamic_fields: [
+                { string_key: 'term', display_label: 'Value', field_type: DynamicField::Type::CONTROLLED_TERM, controlled_vocabulary: 'name' },
+                { string_key: 'is_primary', display_label: 'Is Primary?', field_type: DynamicField::Type::BOOLEAN }
+              ],
+              dynamic_field_groups: [
+                {
+                  string_key: 'role',
+                  display_label: 'Role',
+                  dynamic_fields: [
+                    { string_key: 'term', display_label: 'Value', field_type: DynamicField::Type::CONTROLLED_TERM, controlled_vocabulary: 'name_role' }
+                  ]
+                }
+              ]
+            },
+            {
+              string_key: 'genre',
+              display_label: 'Genre',
+              dynamic_fields: [
+                { string_key: 'term', display_label: 'Value', field_type: DynamicField::Type::CONTROLLED_TERM, controlled_vocabulary: 'genre' }
+              ]
+            }
+          ]
+        }
+      ]
+    }
+  end
+
+  let(:descriptive_metadata) do
+    {
+      'name' => [
+        {
+          'term' => {
+            'prefLabel' => 'something',
+            'uri' => 'temp:4712538a19c162783874c45c4682fddcf247ac031b06a98912d7df0bc43a3a54'
+          },
+          'role' => [
+            {
+              'term' => {
+                'pref_label' => 'author',
+                'uri' => 'http://id.loc.gov/vocabulary/relators/aut'
+              }
+            }
+          ]
+        }
+      ],
+      'genre' => [
+        {
+          'term' => {
+            'uri' => 'http://vocab.getty.edu/aat/300048715',
+            'other_data' => 'something'
+          }
+        }
+      ]
+    }
+  end
+
+  # Load field definitions and create terms.
+  before do
+    Hyacinth::DynamicFieldsLoader.load_fields!(field_definitions)
```

Comment:
Is it worth using `load_vocabularies: true` here to cover the vocabulary creation cases below, and then you can just change the `name_vocab = Vocabulary.create(...` line to `name_vocab = Vocabulary.find_by(string_key: 'name')`?

Reply:
Oh, I see.  You're also making use of the `genre_vocab` and `name_role_vocab` references later.  I guess it doesn't save you much if you have to call find_by on those too.

Reply:
Right!